services/user/Fractals/service/src/tables.rs

```diff
@@ -0,0 +1,573 @@
+            );
+
+            self.create_evaluation(
+                evaluation.registration_starts + interval,
+                evaluation.deliberation_starts + interval,
+                evaluation.submission_starts + interval,
+                evaluation.finish_by + interval,
+                interval,
+            );
+        }
+
+        pub fn set_evaluation_schedule(
+            &mut self,
+            registration: u32,
+            deliberation: u32,
+            submission: u32,
+            finish_by: u32,
+            interval_seconds: u32,
+            force_delete: bool,
+        ) {
+            if self.evaluation_id.is_some() {
+                psibase::services::evaluations::Wrapper::call()
+                    .delete(self.evaluation_id.unwrap(), force_delete);
+            };
+
+            self.create_evaluation(
+                registration,
+                deliberation,
+                submission,
+                finish_by,
+                interval_seconds,
+            );
+        }
+
+        pub fn save(&self) {
+            let table = EvaluationInstanceTable::new();
+            table.put(&self).expect("failed to save");
+        }
+
+        fn scores(&self) -> Vec<Score> {
+            let table = ScoreTable::new();
+
+            table
+                .get_index_pk()
+                .range(
+                    (self.fractal, self.eval_type, AccountNumber::from(0))
+                        ..=(self.fractal, self.eval_type, AccountNumber::from(u64::MAX)),
+                )
+                .collect()
+        }
+
+        pub fn set_pending_scores(&self, pending_score: f32) {
+            self.scores().into_iter().for_each(|mut account| {
+                account.set_pending_score(pending_score);
+            });
+        }
+
+        pub fn award_group_scores(&self, group_number: u32, vanilla_group_result: Vec<u8>) {
+            let group_members = self.users(Some(group_number)).unwrap();
+
+            let fractal_group_result = parse_rank_to_accounts(
+                vanilla_group_result,
+                group_members.into_iter().map(|user| user.user).collect(),
+            );
+
+            for (index, account) in fractal_group_result.into_iter().enumerate() {
+                let level = (6 as usize) - index;
+                Score::get(self.fractal, self.eval_type.into(), account)
+                    .set_pending_score(level as f32);
+            }
+        }
+
+        pub fn save_pending_scores(&self) {
+            self.scores().into_iter().for_each(|mut account| {
+                account.save_pending_score();
+            });
+        }
+    }
+
+    #[table(name = "ScoreTable", index = 3)]
+    #[derive(Default, Fracpack, ToSchema, SimpleObject, Serialize, Deserialize, Debug)]
+    pub struct Score {
+        pub fractal: AccountNumber,
+        pub account: AccountNumber,
+        pub eval_type: EvalTypeU32,
+        pub value: f32,
```

Comment:
Might want to use fixed point arithmetic and store value as u32. Like, a score of 3.5 would be 3500 or 35000.
[Here](https://github.com/gofractally/psibase/pull/1221) is some fixed point ema math